app/javascript/feedback.js

```diff
@@ -0,0 +1,47 @@
+const initFeedback = () => {
+  const prompt = document.querySelector('[data-freetext-feedback="prompt"]');
+  const cancel = document.querySelector('[data-freetext-feedback="cancel"]');
+  const send  = document.querySelector('[data-freetext-feedback="send"]');
+  const yes_button = document.querySelector('#data-satisfaction-feedback-yes');
+  const no_button = document.querySelector('#data-satisfaction-feedback-no');
+
+
+  prompt?.addEventListener('click', () => {
+    showSection("freetext");
+  });
+
+  cancel?.addEventListener('click', () => {
+    showSection("prompt");
+  })
+
+  send?.addEventListener('click', (e) => {
+    const freetextField = document.getElementById("text-field")
+
+    if (freetextField.value === "") {
```

Comment:
Minor bug here: if the user enters ` ` (a space, or several spaces) this validation here will pass, but the back-end `presence` validation will fail, and we'll get a Sentry notification.

A fix would be to check if the string, with all whitespace removed, is empty here. I.e. `freeTextField.value.replace(/\s+/g, '') === ""`

Reply:
of course.